content/blog/2022-07-05-Practical-data-drift-monitoring-with-DVC.md

```diff
@@ -0,0 +1,337 @@
+DVC could be useful for this task. Now, we will demonstrate how the data
+validation, in our case drift detection, is being used in the project.
+
+## Detect drift in the train vs test dataset
+
+First, we will detect covariate shift in the train/test split for ML training.
+We would like to use four months of bike-sharing data (January - April), and we
+would like to know a good strategy for splitting this data into training and
+test datasets. There are many possibilities for how to achieve that. Two
+intuitive strategies would be:
+
+- Use January-March data for training and April data for testing.
+- Take all months and randomly split the data.
+
+We have set the split strategy as one of the parameters in `params.yaml`, we can
+compare them as follows
+
+```bash
+$ dvc exp run -S data_preparation.data_split=random -n data-split-random
+...
+$ dvc exp run -S data_preparation.data_split=month -n data-split-month
+...
+$ dvc plots diff data-split-month data-split-random --targets plots/temp_feature_distribution.png
+```
+
+Below you can see histogram plots that compare two data splits. For simplicity,
+we show the plots of value distributions for a single feature, `temp` (
+temperature). On the horizontal axis is the value of the `temp` feature. It
+ranges from 0 to 1 because it is normalized in the dataset. On the vertical axis
+is a relative prevalence of the bin. This prevalence is also normalized, meaning
+that the sum of all bins for one dataset is equal to 1.
+
+![Value distributions of 'temp' feature](/uploads/images/2022-07-05/data_distribution.png)
+
+We can see that the covariate shift detector also confirms a drift between the
+training and test datasets in the case of the “month” split strategy.
+
+## Detect drift in the day patterns
+
+A concept drift occurs when there is a change in the pattern that the ML model
+learns and new data. To learn and detect the pattern, we need to use more
+training data, in our case one extra year. With this historical data, we have
+learned that there are two bike-rental patterns, one for a working day and one
+for a non-working day.
+
+The most important configuration parameter is for us the `latent_dim` , which
+represents the latent dimension of the Seq2Seq encoder and decoder, we will
+experiment with three different values for each detector. In DVC, we do it as
+follows.
+
+```bash
+$ dvc exp run --queue -S detector.concept_drift.latent_dim_working=24 -S detector.concept_drift.latent_dim_noworking=6
+$ dvc exp run --queue -S detector.concept_drift.latent_dim_working=48 -S detector.concept_drift.latent_dim_noworking=12
+$ dvc exp run --queue -S detector.concept_drift.latent_dim_working=96 -S detector.concept_drift.latent_dim_noworking=24
+$ dvc exp run --run-all
+```
+
+A comparison of the experiments in the VS Code extension looks as follows.
+
+![Experiments comparison in VS Code](/uploads/images/2022-07-05/vs_code_experiments_comparison.png)
```

Comment:
I think a ```dvctable``` block would look better than the VSCode extension screenshot and just mention `dvc exp show`? 

The mention to VSCode extension looks kind of disconnected from the rest as we are showing CLI snippets for running the experiments